Declarations/Constants.ts

```diff
@@ -0,0 +1,100 @@
+import Contracts = require("../Declarations/Contracts")
+
+export const QuickPulseConfig = {
+    method: "POST",
+    time: "x-ms-qps-transmission-time",
+    subscribed: "x-ms-qps-subscribed"
+};
+
+export enum QuickPulseCounter {
+    // Memory
+    COMMITTED_BYTES= "\\Memory\\Committed Bytes",
+
+    // CPU
+    PROCESSOR_TIME= "\\Processor(_Total)\\% Processor Time",
+
+    // Request
+    REQUEST_RATE= "\\ApplicationInsights\\Requests\/Sec",
+    REQUEST_FAILURE_RATE= "\\ApplicationInsights\\Requests Failed\/Sec",
+    REQUEST_DURATION= "\\ApplicationInsights\\Request Duration",
+
+    // Dependency
+    DEPENDENCY_RATE= "\\ApplicationInsights\\Dependency Calls\/Sec",
+    DEPENDENCY_FAILURE_RATE= "\\ApplicationInsights\\Dependency Calls Failed\/Sec",
+    DEPENDENCY_DURATION= "\\ApplicationInsights\\Dependency Call Duration",
+
+    // Exception
+    EXCEPTION_RATE= "\\ApplicationInsights\\Exceptions\/Sec"
+}
+
+export enum PerformanceCounter {
+    // Memory
+    PRIVATE_BYTES= "\\Process(??APP_WIN32_PROC??)\\Private Bytes",
+    AVAILABLE_BYTES= "\\Memory\\Available Bytes",
+
+    // CPU
+    PROCESSOR_TIME= "\\Processor(_Total)\\% Processor Time",
+    PROCESS_TIME= "\\Process(??APP_WIN32_PROC??)\\% Processor Time",
+
+    // Requests
+    REQUEST_RATE= "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Requests/Sec",
+    REQUEST_DURATION= "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Request Execution Time"
+};
+
+/**
+ * Map a PerformanceCounter/QuickPulseCounter to a QuickPulseCounter. If no mapping exists, mapping is *undefined*
+ */
+export const PerformanceToQuickPulseCounter: {[key: string]: QuickPulseCounter} = {
```

Comment:
You might be able to make the type of key `PerformanceCounter | QuickPulseCounter`

Reply:
Added `Contracts.TelemetryTypeKeys` and `Contracts.TelemetryTypeValues`